test/algorithms/set_operations/set_ops_areal_areal.cpp

```diff
@@ -0,0 +1,218 @@
+    {
+        BOOST_CHECK_MESSAGE(bgeo::is_valid(geometry1),
+            "Case: " << name << " geometry1 is not valid");
+        BOOST_CHECK_MESSAGE(bgeo::is_valid(geometry2),
+            "Case: " << name << " geometry2 is not valid");
+        BOOST_CHECK_MESSAGE(bgeo::is_valid(result_union),
+            "Case: " << name << " union is not valid");
+        BOOST_CHECK_MESSAGE(bgeo::is_valid(result_intersection),
+            "Case: " << name << " intersection is not valid");
+        if (settings.test_validity_diff1)
+        {
+            BOOST_CHECK_MESSAGE(bgeo::is_valid(diff1),
+                "Case: " << name << " difference (a-b) is not valid");
+        }
+        if (settings.test_validity_diff2)
+        {
+            BOOST_CHECK_MESSAGE(bgeo::is_valid(diff2),
+                "Case: " << name << " difference (b-a) is not valid");
+        }
+        if (settings.test_validity_diff_sym)
+        {
+            BOOST_CHECK_MESSAGE(bgeo::is_valid(sym_diff),
+                "Case: " << name << " symmetric difference is not valid");
+        }
+    }
+}
+
+template <typename T>
+void test(std::string const& name, std::string const& wkt1, std::string const& wkt2,
+    ut_settings const& settings)
+{
+	using point_t = bgeo::model::point<T, 2, bgeo::cs::cartesian>;
+	using polygon_t = bgeo::model::polygon<point_t>;
+	using mp_t = bgeo::model::multi_polygon<polygon_t>;
+
+    constexpr auto multipolygon_tag = "MULTIPOLYGON";
+    bool const is_polygon1 = ! boost::icontains(wkt1, multipolygon_tag);
+    bool const is_polygon2 = ! boost::icontains(wkt2, multipolygon_tag);
+
+    if (is_polygon1 && is_polygon2)
+    {
+        test_detail<T, mp_t, polygon_t, polygon_t>(name, wkt1, wkt2, settings);
+    }
+    else if (! is_polygon1 && ! is_polygon2)
+    {
+        test_detail<T, mp_t, mp_t, mp_t>(name, wkt1, wkt2, settings);
+    }
+    else if (is_polygon1)
+    {
+        test_detail<T, mp_t, polygon_t, mp_t>(name, wkt1, wkt2, settings);
+    }
+    else
+    {
+        test_detail<T, mp_t, mp_t, polygon_t>(name, wkt1, wkt2, settings);
+    }
+}
+
+void test_all(std::string const& name, std::string const& wkt1, std::string const& wkt2,
+    ut_settings const& settings = ut_settings())
+{
+    test<double>(name, wkt1, wkt2, settings);
+}
+
+#define TEST_CASE(caseid) (test_all(#caseid, caseid[0], caseid[1]))
+#define TEST_CASE_WITH(caseid, index1, index2, settings) (test_all(#caseid "_" #index1 "_" #index2, caseid[index1], caseid[index2], settings))
+
+int test_main(int, char* [])
+{
+    TEST_CASE(issue_1222);
+    TEST_CASE_WITH(issue_1226, 0, 1, ut_settings().ignore_validity_diff());
+    TEST_CASE(issue_1231);
+    TEST_CASE(issue_1244);
+    TEST_CASE_WITH(issue_1288, 0, 1, ut_settings().ignore_validity_diff());
+    TEST_CASE_WITH(issue_1288, 0, 2, ut_settings());
+    TEST_CASE(issue_1293);
+    TEST_CASE_WITH(issue_1295, 0, 1, ut_settings().ignore_validity_diff());
+    TEST_CASE(issue_1299);
+
+    TEST_CASE(issue_1326);
```

Comment:
So, in summary, this one-line replace four lines with areas, spread over 3 files:

```
   TEST_UNION(issue_1326, 1, 0, -1, 23.201);
   TEST_UNION_REV(issue_1326, 1, 0, -1, 23.201);
   TEST_INTERSECTION(issue_1326, 1, -1, 16.4844);
   TEST_DIFFERENCE(issue_1326, 3, 6.7128537626409130468, 6, 0.00372806966532758478, 9);
```